CMakeLists.txt

```diff
@@ -39,10 +39,8 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
         # Additional setup for coverage
         # https://gcovr.com/en/stable/guide/compiling.html#compiler-options
         #
-        if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Please add a new flag for coverage (I think coverage should be disabled by default), otherwise I can not build with Microsoft compiler 🥺

Reply:
What's the compiler ID for the compiler you're using?

Reply:
It would be `MSVC`, but as far as I know the stuff we are adding also does not work with `Clang` (needs different compiler args) and probably others. That's why I have suggested to make it optional.

Reply:
Clang accepts almost all gnu flags, as far as I know... Certainly it works for these flags, otherwise this fix wouldn't work.

Reply:
I read somewhere that it needs additional flags, but I cannot find it anymore. In that case disabling for msvc is enough then.